src/whylogs/app/logger.py

```diff
@@ -75,45 +80,82 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+
+    @property
+    def profile(self,):
+        """
+        :return: the last backing dataset profile
+        :rtype: DatasetProfile
+        """
+        return self._profiles[-1]["full_profile"]
+
     @property
-    def profile(self):
+    def segmented_profiles(self,)->DatasetProfile:
         """
         :return: the last backing dataset profile
         :rtype: DatasetProfile
         """
-        return self._profiles[-1]
+        
+        return self._profiles[-1]["segmented_profiles"]
+
+    def set_segments(self, segments: Union[List[List[Dict[str,str]]],List[str]]) -> None:
+        if segments:
+            if all(isinstance(elem, str) for elem in segments):
+                self.segment_type = "keys"
+                self.segments= segments
+            else:
+                self.segments=segments
+                self.segment_type = "set"
+        else:
+            self.segments=None
+            self.segment_type=None
+
+    def intialize_profiles(self, 
+        dataset_timestamp: Optional[datetime.datetime] = datetime.datetime.now(datetime.timezone.utc)) -> None:
+        
 
+        full_profile=None
+        if self.full_profile_check():
+            full_profile = DatasetProfile(
```

Comment:
we call it `overall` - let's keep the terminology consistent

Reply:
+1

Reply:
actually I find `overall` is a bit confusing versus full_profile and segmented_profile.